PZNS_Framework/media/lua/client/03_mod_core/PZNS_NPCSurvivor.lua

```diff
@@ -1,61 +1,204 @@
+        _maxRelation = 50,
+        anonymousMap = {},
+        affection = 50,
         ------ IsoPlayer Spawning Related below ------
-        isAlive = true,                         -- WIP - Technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        isSpawned = true,                       -- WIP - Technically part of IsoPlayer... but "isExistInTheWorld()" seems very inconsistent...
-        forename = "",                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        surname = "",                           -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        isFemale = false,                       -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareX = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareY = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        squareZ = nil,                          -- Cows: Placeholder; technically part of IsoPlayer; used when PZNS needs to use it before IsoPlayer is loaded.
-        npcIsoPlayerObject = npcIsoPlayerObject -- Cows: objects cannot be saved to moddata...
+        isAlive = true,
+        isSpawned = true,
+        forename = "",
+        surname = "",
+        isFemale = false,
+        squareX = nil,
+        squareY = nil,
+        squareZ = nil,
+        npcIsoPlayerObject = npcIsoPlayerObject
     };
 
+    setmetatable(npcSurvivor, self);
+    self.__index = self;
+
     return npcSurvivor;
 end
+
+---Assigns groupID to NPC
+---@param groupID groupID
+function NPC:setGroupID(groupID)
+    self.groupID = groupID
+end
+
+---Unassign groupID from NPC
+function NPC:unsetGroupID()
+    self.groupID = nil
+end
+
+---comments
+---@param mpPlayerID integer
+function NPC:setPlayerNum(mpPlayerID)
+    local requestedIsoPlayer = getSpecificPlayer(mpPlayerID)
+    assert(requestedIsoPlayer, "Can't find specific player, num: " .. tostring(mpPlayerID))
+    local msg = string.format("IsoPlayer for mpPlayerID: %s does not match npcIsoPlayerObject. survivorID: %s",
+        mpPlayerID, self.survivorID)
+    assert(requestedIsoPlayer == self.npcIsoPlayerObject, msg)
+    self.mpPlayerID = mpPlayerID
+end
+
+--region relations
+
+---Change relation with `otherSurvivorID` by `diff`
+---@param otherSurvivorID survivorID survivor to change relation to
+---@param diff integer change amount
+---@return boolean isNewRelation whether this relation is new (first meet)
+function NPC:changeRelation(otherSurvivorID, diff)
+    --TODO: periodic(?) cleanings for nil NPC (dead etc)
+    local relation = self.relationsMap[otherSurvivorID]
```

Comment:
One of my worries with NPC-to-NPC relations is yes, the amount of background processing that may be needed... 
The current NPC-to-Player is n(x) operations, what will the future NPC-to-NPC operations look like (n^2)? 
I suppose it isn't an issue if it's only run during a set time... but there will need to be multiple checks for dead/living and possibly whether or not the NPC is loaded on screen or in a loaded square..

Reply:
Well, as I mentioned, this was a basic idea without much thought put into it.